openedx/core/djangoapps/credit/verification_access.py

```diff
@@ -0,0 +1,202 @@
+                    "associated in-course-reverification checkpoint does not exist."
+                ),
+                partition.id, course_key
+            )
+
+        # Preserve non-verified partitions without modifying them.
+        else:
+            results.append(partition)
+            log.info(
+                (
+                    "Preserved partition %s in course %s because it is not "
+                    "using the verification partition scheme."
+                ),
+                partition.id, course_key
+            )
+
+    return results
+
+
+def _set_verification_partitions(course_key, icrv_blocks):
+    """
+    Create or update user partitions in the course.
+
+    Ensures that each ICRV block in the course has an associated user partition
+    with the groups ALLOW and DENY.
+
+    Arguments:
+        course_key (CourseKey): Identifier for the course.
+        icrv_blocks (list of XBlock): In-course reverification blocks, e.g. reverification checkpoints.
+
+    Returns:
+        list of UserPartition
+    """
+    scheme = UserPartition.get_scheme(VERIFICATION_SCHEME_NAME)
+    if scheme is None:
+        log.error("Could not retrieve user partition scheme with ID %s", VERIFICATION_SCHEME_NAME)
+        return []
+
+    course = modulestore().get_course(course_key)
+    if course is None:
+        log.error("Could not find course %s", course_key)
+        return []
+
+    partition_id_for_location = {
+        p.parameters["location"]: p.id
+        for p in course.user_partitions
+        if p.scheme == scheme and "location" in p.parameters
+    }
+
+    partitions = []
+    for block in icrv_blocks:
+        partition = UserPartition(
+            id=partition_id_for_location.get(
+                unicode(block.location),
+                _unique_partition_id(course)
+            ),
+            name=block.related_assessment,
+            description=u"Verification checkpoint at {}".format(block.related_assessment),
+            scheme=scheme,
+            parameters={"location": unicode(block.location)},
+            groups=[
+                Group(scheme.ALLOW, "Completed verification at {}".format(block.related_assessment)),
+                Group(scheme.DENY, "Did not complete verification at {}".format(block.related_assessment)),
+            ]
+        )
+        partitions.append(partition)
+
+        log.info(
+            (
+                "Configured partition %s for course %s using a verified partition scheme "
+                "for the in-course-reverification checkpoint at location %s"
+            ),
+            partition.id,
+            course_key,
+            partition.parameters["location"]
+        )
+
+    # Preserve existing, non-verified partitions from the course
+    # Mark partitions for deleted in-course reverification as disabled.
+    course.user_partitions = partitions + _other_partitions(course.user_partitions, partitions, course_key)
```

Comment:
Instead of modifying the attribute directly here, can we go for `get_partitions_for_scheme()` and `set_partitions_for_scheme` methods on `CourseModule`?